task01/include/Vector.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include <iostream>
+
+
+class Vector {
+public:
+    using iterator = int*;
+
+    /* constructor section */
+    Vector(const Vector& other);
+    Vector(Vector&& other) noexcept;
+    Vector(std::size_t size = 0, int value = 0);
+
+    Vector& operator=(const Vector& other);
+    Vector& operator=(Vector&& other) noexcept;
+
+    int& operator[](std::size_t index) { return m_array[index]; }
```

Comment:
You are correct that the `operator[]` signature returns `T&`, but specifically for a trivially-copyable type like int there is no need to return a reference, the compiler will just omit it and copy the value